### PR TITLE
Hotfix for health monitor plans

### PR DESCRIPTION
**Pull request by kmdalal**
Note: Jira OW-1223 was created for this work after the branch was already pushed, so leaving it be and have noted this in the Jira as well.

### Summary

- Adds correct behavior for no power situation.
- Corrects and elaborates plan comments.
- Corrects if-then groupings to avoid unnecessary checks.
- Adds the `endif` keyword for better clarity and maintenance.
- Improves file formatting.

### Test

1. Build this branch with all other repos on their `master` branches.
2. Start any simulator world.
3. Start the executive with `roslaunch ow_plexil ow_exec.launch fault_dependencies_file:="FaultDependenciesModel.xml"`
4. Start the `HealthMonitorDemo` plan.

The arm will start unstowing and stowing repeatedly.  Now inject and clear a variety of faults as instructed below.  In addition to the faults being reported in both terminals, you'll see `operable` and `inoperable` messages in the executive terminal.

Note that you may have to wait a few seconds to see the expected behavior, due to action timeouts, and also that there may occasional out-of-order messages due to interleaving of concurrent nodes.

1. Inject any arm fault(s).  The arm should stop and be reported inoperable, and the camera should start taking pictures.
2. Inject the camera fault.  The camera will be reported inoperable, but continue to attempt images.
3. Inject a `low_state_of_charge` or `thermal_failure` power fault.   The camera should stop altogether and Power should be reported inoperable.
5. Clear the power fault.  The behavior of (2) should return.
6. Clear the camera fault.  The camera should be reported operable and start operating normally again.
7. Clear the arm fault(s).  The arm should be reported operable and start moving, and the camera should stop.
8. If you like, inject and clear Arm, Camera, and Power faults in any order and see that behavior conforms to the comments in the `HealthMonitorDemo` plan.

### Review Suggestion

One technical review is sufficient, and requesting @AstroStucky for this.  @mogumbo feel free to review or just provide the second approval needed for this hotfix.

### CODE REVIEW

**Review by mogumbo — APPROVED** [review]

**Comment by AstroStucky** [comment]
Should the antenna pan and tilt to look at the end-effector before taking a picture like we discussed, or is that still to come in another PR?

**Comment by kmdalal** [comment]
> Should the antenna pan and tilt to look at the end-effector before taking a picture like we discussed, or is that still to come in another PR?

That's future work.  It's an enhancement, which I didn't think suitable for a hotfix.

File: ow_plexil/src/plans/HealthMonitor.plp [hunk 1]
```diff
@@ -44,61 +50,69 @@ HealthMonitor:
...
+    if !ArmOperable && PrevArm {
+      if Lookup(IsFaulty("Arm")) {
+        log_warning("Health Monitor: Arm is Inoperable due to the local fault(s): ",
+                    Lookup(ActiveFaults("Arm")));
       }
-      else{
+      else {
         log_warning("Health Monitor: Arm is Inoperable due to non-local fault(s).");
       }
       PrevArm = false;
```
Comment (AstroStucky) [inline ow_plexil/src/plans/HealthMonitor.plp:61]: I am unclear on what the `Prev*` variables represent. They could use a comment explaining them or a more descriptive name.

**Review by AstroStucky — APPROVED** [review]
Test passed without issues. 

In addition to my optional code comment, I wanted to make a suggestion about the HealthMonitor LibraryAction. It repeats 4 identical if-else statement patterns with only Booleans and Strings differing. A more versatile LibraryAction would be one that captures only this if-else statement pattern with InOut Booleans and In String parameters. This is assuming PLEXIL InOut variables behave as non-const references as I believe they do.

File: ow_plexil/src/plans/HealthMonitor.plp (hunk 1 above)
Comment (kmdalal) [reply ow_plexil/src/plans/HealthMonitor.plp]: I had the same feeling.  Though verbose, I have renamed `PrevArm` to `ArmPreviouslyWorking` to make the meaning clear.  I renamed the other flags accordingly and added a comment about them.

**Comment by kmdalal** [comment]
> I wanted to make a suggestion about the HealthMonitor LibraryAction. It repeats 4 identical if-else statement patterns with only Booleans and Strings differing. A more versatile LibraryAction would be one that captures only this if-else statement pattern with InOut Booleans and In String parameters. This is assuming PLEXIL InOut variables behave as non-const references as I believe they do.

Good observation, and yes, the plan could be refactored this way.  I've left this for future work given the other fundamental improvements that could be made.  I added a comment at the top saying this is a simple prototype.

**Pull request merged**